### PR TITLE
New custom basemap category for TileJSON basemaps

### PR DESCRIPTION
**Pull request by viddo**
Implements the last category for the custom basemaps modal.
screen shot 2015-05-28 at 16 53 11
The new category essentially has the same look&feel as the XYZ tab.

Also remove link to old modal, since all categories are now in pair with it.

~~TMS option to be added separately, once the [PR for adding TMS option to XYZ URLs (#3773)](https://github.com/CartoDB/cartodb/pull/3773) is merged.~~ nevermind, blackout on my part… the TMS setting is defined by the TileJSON result so there's no option to set by the user, just passing the value properly when adding the basemap layer (solved in commit 5871974).

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3967/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3969/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3970/))

**Comment by viddo** [comment]
Please review @xavijam and @saleiva 
Deployed on staging ded06 so you can give it try if you feel like it.

File: lib/assets/javascripts/cartodb/common/dialogs/add_custom_basemap/tabs.jst.ejs [hunk 1]
```diff
@@ -13,10 +13,6 @@
                   </button>
                 </li>
               <% }) %>
-              <%/* TODO: Remove once all tabs are implemented */%>
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/common/dialogs/add_custom_basemap/tabs.jst.ejs]: :) yay!

**Comment by xavijam** [comment]
Looks good!

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3986/))

**Pull request merged**